src/main/java/seedu/address/logic/parser/AddTaskParser.java

```diff
@@ -0,0 +1,260 @@
+        String dueTime;
+    }
+
+    private static ParseResult parseArguments(String str) throws IllegalValueException {
+        final ParseResult result = new ParseResult();
+        final ArrayList<String> args = splitArgs(str);
+
+        String mediumValueStartingOrDueDate = null;
+        String mediumValueStartingOrDueTime = null;
+
+        // name
+        if (args.isEmpty()) {
+            throw new IllegalValueException("expected name");
+        }
+        result.name = args.remove(0);
+
+        // priority (optional)
+        if (args.isEmpty()) {
+            result.type = "float";
+            return result;
+        }
+        if (isPriority(args.get(0))) {
+            result.type = "float";
+            Matcher matcher = PRIORITY_PATTERN.matcher(args.remove(0));
+            matcher.find();
+            result.priority = matcher.group("priority");
+            if (!args.isEmpty()) {
+                throw new IllegalValueException("too many arguments");
+            }
+            return result;
+        }
+
+        // starDate or dueDate (optional)
+        if (args.isEmpty()) {
+            return result;
+        }
+        if (isDate(args.get(0))) {
+            mediumValueStartingOrDueDate = args.remove(0);
+        }
+
+        // startTime or dueTime (optional)
+        if (args.isEmpty()) {
+            if (mediumValueStartingOrDueDate != null) {
+                result.type = "deadline";
+                result.dueDate = mediumValueStartingOrDueDate;
+            }
+            return result;
+        }
+        if (isTime(args.get(0))) {
+            mediumValueStartingOrDueTime = args.remove(0);
+        }
+
+        // check if there is a "to" inside the command
+        if (args.isEmpty()) {
+            if (mediumValueStartingOrDueDate != null) {
+                result.type = "deadline";
+                result.dueDate = mediumValueStartingOrDueDate;
+            }
+            if (mediumValueStartingOrDueTime != null) {
+                result.type = "deadline";
+                result.dueTime = mediumValueStartingOrDueTime;
+            }
+            return result;
+        }
+        if (isKeywordTo(args.get(0))) {
+            args.remove(0);
+            result.type = "event";
+            result.startDate = mediumValueStartingOrDueDate;
+            result.startTime = mediumValueStartingOrDueTime;
+        }
+
+        // endDate (optional)
+        if (args.isEmpty()) {
+            return result;
+        }
+        if (isDate(args.get(0))) {
+            result.endDate = args.remove(0);
+        }
+
+        // endTime (optional)
+        if (args.isEmpty()) {
+            return result;
+        }
+        if (isTime(args.get(0))) {
+            result.endTime = args.remove(0);
+        }
+
+        if (!args.isEmpty()) {
+            throw new IllegalValueException("wrong arguments given, please check the number and the order of the arguments");
+        }
+        return result;
```

Comment:
Ugh, I thought that we were going to get a reusable parser that we can use for all commands (and implement new ones in the future), not coding everything manually...